### PR TITLE
doc: document bluestore compression settings

### PR DESCRIPTION
**Pull request by tchaikov**
Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

File: doc/rados/configuration/bluestore-config-ref.rst [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+BlueStore Config Reference
+==========================
+
+Inline Compression
+==================
+
+BlueStore supports inline compression using snappy, zlib, or LZ4. Please note,
+the lz4 compression plugin is not distributed by the official release.
+
+``bluestore compression algorithm``
```
Comment (xiexingguo) [inline doc/rados/configuration/bluestore-config-ref.rst]: bluestore_compression_algorithm?

File: doc/rados/configuration/bluestore-config-ref.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
+==========================
+BlueStore Config Reference
+==========================
+
+Inline Compression
+==================
+
+BlueStore supports inline compression using snappy, zlib, or LZ4. Please note,
+the lz4 compression plugin is not distributed by the official release.
```
Comment (theanalyst) [inline doc/rados/configuration/bluestore-config-ref.rst]: distributed in the official release?

File: doc/rados/configuration/bluestore-config-ref.rst [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+``bluestore compression algorithm``
+
+:Description: The default compressor to use (if any) if the per-pool property
+              ``compression_mode`` is not set. Note that zstd is *not*
+              recommended for blustore due to high CPU overhead when
```
Comment (theanalyst) [inline doc/rados/configuration/bluestore-config-ref.rst]: bluestore

File: doc/rados/configuration/bluestore-config-ref.rst [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+              compressing small amounts of data.
+:Type: String
+:Required: No
+:Valid Settings: ``lz4``, ``snappy``, ``zlib``, ``zstd``
+:Default: ``snappy``
+
+``bluestore_compression_mode``
+
+:Description: The default policy for using compression if the per-pool property
+              ``compressoin_mode`` is not set. ``none`` means never use
```
Comment (theanalyst) [inline doc/rados/configuration/bluestore-config-ref.rst]: typo compression_mode

File: doc/rados/configuration/bluestore-config-ref.rst [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+              ``force`` means use compression under all circumstances even if
+              the clients hint that the data is not compressible.
+:Type: String
+:Required: No
+:Valid Settings: ``none``, ``passive``, ``aggressive``, ``force``
+:Default: ``none``
+
+``bluestore_compression_min_blob_size``
+
+:Description: Chunks smaller thank this are never compressed.
```
Comment (theanalyst) [inline doc/rados/configuration/bluestore-config-ref.rst]: than

File: doc/rados/configuration/bluestore-config-ref.rst [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+Inline Compression
+==================
+
+BlueStore supports inline compression using snappy, zlib, or LZ4. Please note,
+the lz4 compression plugin is not distributed by the official release.
+
+``bluestore compression algorithm``
+
+:Description: The default compressor to use (if any) if the per-pool property
+              ``compression_mode`` is not set. Note that zstd is *not*
```
Comment (xiexingguo) [inline doc/rados/configuration/bluestore-config-ref.rst]: This should be compression_algorithm instead.

File: doc/rados/configuration/bluestore-config-ref.rst (hunk 5 above)
Comment (xiexingguo) [inline doc/rados/configuration/bluestore-config-ref.rst]: s/thank/than/

**Comment by tchaikov** [comment]
@theanalyst @xiexingguo fixed and repushed.

File: doc/rados/configuration/bluestore-config-ref.rst (hunk 1 above)
Comment (tchaikov) [inline doc/rados/configuration/bluestore-config-ref.rst]: will use spaces for consistency.

File: doc/rados/configuration/bluestore-config-ref.rst [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+:Description: Chunks smaller than this are never compressed.
+              The per-pool property ``compression_min_blob_size`` overrides
+              this setting.
+
+:Type: Unsigned Integer
+:Required: No
+:Default: 0
+
+``bluestore_compression_min_blob_size_hdd``
```
Comment (xiexingguo) [inline doc/rados/configuration/bluestore-config-ref.rst]: Keep pace with others, then?

File: doc/rados/configuration/bluestore-config-ref.rst [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+:Description: Default value of ``bluestore_compression_min_blob_size``
```
Comment (xiexingguo) [inline doc/rados/configuration/bluestore-config-ref.rst]: And here

File: doc/rados/configuration/bluestore-config-ref.rst (hunk 7 above)
Comment (tchaikov) [inline doc/rados/configuration/bluestore-config-ref.rst]: should be addressed by the latest change.

**Comment by xiexingguo** [comment]
looks good to me,  a very small me as an ESL speaker:-)

**Pull request merged**